pmlc/compiler/program.cc

```diff
@@ -149,8 +149,19 @@ void Program::compile(StringRef targetName, bool collectPasses,
                         /*out=*/llvm::errs());
   }
 
+  auto begOpts = targetNameAndOptions.find('{');
+  auto targetName = targetNameAndOptions.substr(0, begOpts);
+  auto targetOptions = targetNameAndOptions.substr(begOpts);
+
+  // if target options are specified
+  if (!targetOptions.empty()) {
+    // trim off curly braces
+    auto endOpts = targetOptions.find('}');
+    targetOptions = targetOptions.substr(1, endOpts - 1);
+  }
```

Comment:
This is the only part that gives me pause. I wonder if we can use the existing parse pipeline functionality of MLIR for this.

Reply:
This code separates target name from options.  E.g. PLAIDML_TARGET="mytarget{myoption1=true myoption2=7}" you get two strings targetName = "mytarget" and targetOptions = "myoption1=true myoption2=7".  I did not find an MLIR utility that does this.  I do parse the targetOptions with an MLIR utility called `createFromString`.